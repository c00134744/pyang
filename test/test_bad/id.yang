module id {
  namespace "urn:id";
  prefix id;

  import "crypto-base" {
    prefix "crypto";
  }

  feature x { // error: circular
    if-feature y;
  }

  feature y {
    if-feature x;
  }

  feature z {
    if-feature x2; // error: x2 does not exist
  }

  feature z; // error: already defined

  leaf foo {
    if-feature x;
    type string;
  }

  leaf foo2 {
    if-feature x2; // error: not defined
    type string;
  }

  identity x { // error: circular
    base y;
  }

  identity y {
    base x;
  }

  identity y; // error: already defined

  identity des {
    base "crypto:crypto-alg";
    description "DES crypto algorithm";
  }
  
  identity des3 {
    base "crypto:crypto-alg";
    description "Triple DES crypto algorithm";
  }
  
  identity des4 {
    base "crypto:crypto-alg2"; // error: not defined
    description "Triple DES crypto algorithm";
  }
  
  identity des5 {
    base "crypto-alg2"; // error: not defined
    description "Triple DES crypto algorithm";
    base "crypto-alg2"; // error: syntax
  }
}
