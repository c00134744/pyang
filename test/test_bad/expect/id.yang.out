id.yang:9: error: CIRCULAR_DEPENDENCY
id.yang:18: error: FEATURE_NOT_FOUND
id.yang:21: error: FEATURE_ALREADY_DEFINED
id.yang:29: error: FEATURE_NOT_FOUND
id.yang:33: error: CIRCULAR_DEPENDENCY
id.yang:41: error: IDENTITY_ALREADY_DEFINED
id.yang:54: error: IDENTITY_NOT_FOUND
id.yang:59: error: IDENTITY_NOT_FOUND
id.yang:61: error: UNEXPECTED_KEYWORD
